src/matroskavideo.cpp

```diff
@@ -0,0 +1,800 @@
+  uint64_t tag = b0 << ((size - 1) * 8);
+  for (size_t i = 1; i < size; ++i) {
+    tag |= static_cast<uint64_t>(buf[i]) << ((size - i - 1) * 8);
+  }
+
+  return tag;
+}
+
+/*!
+    @brief Function used to convert buffer data into numerical information,
+        information stored in BigEndian format
+ */
+[[nodiscard]] uint64_t returnValue(const byte* buf, size_t size) {
+  uint64_t ret = 0;
+  for (size_t i = 0; i < size; ++i) {
+    ret |= static_cast<uint64_t>(buf[i]) << ((size - i - 1) * 8);
+  }
+
+  return ret;
+}
+
+}  // namespace Internal
+}  // namespace Exiv2
+
+namespace Exiv2 {
+
+using namespace Exiv2::Internal;
+
+MatroskaVideo::MatroskaVideo(BasicIo::UniquePtr io) : Image(ImageType::mkv, mdNone, std::move(io)) {
+}  // MatroskaVideo::MatroskaVideo
+
+std::string MatroskaVideo::mimeType() const {
+  return "video/matroska";
+}
+
+void MatroskaVideo::writeMetadata() {
+}
+
+void MatroskaVideo::readMetadata() {
+  if (io_->open() != 0)
+    throw Error(ErrorCode::kerDataSourceOpenFailed, io_->path(), strError());
+
+  // Ensure that this is the correct image type
+  if (!isMkvType(*io_, false)) {
+    if (io_->error() || io_->eof())
+      throw Error(ErrorCode::kerFailedToReadImageData);
+    throw Error(ErrorCode::kerNotAnImage, "Matroska");
+  }
+
+  IoCloser closer(*io_);
+  clearMetadata();
+  continueTraversing_ = true;
+  height_ = width_ = 1;
+
+  xmpData_["Xmp.video.FileName"] = io_->path();
+  xmpData_["Xmp.video.FileSize"] = (double)io_->size() / (double)1048576;
+  xmpData_["Xmp.video.MimeType"] = mimeType();
+
+  while (continueTraversing_)
+    decodeBlock();
+
+  aspectRatio();
+}
+
+void MatroskaVideo::decodeBlock() {
+  byte buf[8];
+  io_->read(buf, 1);
+
+  if (io_->eof()) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  uint32_t sz = findBlockSize(buf[0]);  // 0-8
+  if (sz > 0)
+    io_->read(buf + 1, sz - 1);
+
+  const MatroskaTags* mt = findTag(matroskaTags, returnTagValue(buf, sz));
+
+  if (!mt) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  if (mt->first == 0xc53bb6b || mt->first == 0xf43b675) {
+    continueTraversing_ = false;
+    return;
+  }
+
+  bool skip = find(compositeTagsList, mt->first) != 0;
+  bool ignore = find(ignoredTagsList, mt->first) != 0;
+
+  io_->read(buf, 1);
+  sz = findBlockSize(buf[0]);  // 0-8
+
+  if (sz > 0)
+    io_->read(buf + 1, sz - 1);
+  uint64_t size = returnTagValue(buf, sz);
+
+  if (skip && !ignore)
+    return;
+
+  const uint64_t bufMinSize = 200;
+
+#ifndef SUPPRESS_WARNINGS
+  if (!ignore && size > bufMinSize) {
+    EXV_WARNING << "Size " << size << " of Matroska tag 0x" << std::hex << mt->first << std::dec << " is greater than "
+                << bufMinSize << ": ignoring it.\n";
+  }
+#endif
+  if (ignore || size > bufMinSize) {
+    io_->seek(size, BasicIo::cur);
+    return;
+  }
+
+  DataBuf buf2(bufMinSize + 1);
+  std::fill(buf2.begin(), buf2.end(), 0x0);
+  size_t s = static_cast<size_t>(size);
```

Comment:
[question] Why do you need to do the casting here? Could not you work with `size_t` all the time?  This would avoid the need of casts.

Reply:
done here 2da2a0a8d360decd8afb688f7b751e6395942f66